BNP12

trappmartin opened the pull request:
**This PR aims to showcase Turing for implementing BNP models.**

I plan to implement the following example models for BNP12:

- [x] CRP mixture model on Iris
- [x] topic model
- [x] CrossCat model
- [x] SMC for lazy RPM mixture models (http://approximateinference.org/2017/accepted/Bloem-ReddyEtAl2017.pdf)
- [ ] HDP-HMM as extension of https://turing.ml/tutorials/4-bayeshmm/
- [ ] infinite warped mixtures -> Stheno (GPs not NN) -- not sure I will do this as the integration of Stheno is still wip

@mlomeli1 and @yebai : Do you know other examples that are more relevant / interesting for an statistician audience? Or should we stick to those?

And do you know of any interesting datasets we could use?

yebai commented:
> @mlomeli1 and @yebai : Do you know other examples that are more relevant / interesting for an statistician audience? Or should we stick to those?

HDP-HMM would be a cool example; it would be nice to try HMC (mixture parameters) + PG (component assignments or partitioning structure) for inference on some of these models.

mlomeli1 commented:
Hi @trappmartin I think your examples are ok for the poster. I would like to see Yee Whye and Stefano's reuse algorithm for the marginalised sampler in Turing 'MCMC for Normalized Random Measure Mixture Models' and the lazy sampling from http://approximateinference.org/2017/accepted/Bloem-ReddyEtAl2017.pdf . Emile had some stuff in Turing for the last one. I have some Matlab code for the Reuse algorithm for the sigma-Stable Poisson-Kingman class of mixture models, let me know if you want to check it out.

trappmartin commented:
@mlomeli1 could you send me a link to the Reuse algorithm (paper) and your Matlab code. I'll have a look and would add it if possible. I updated the list and replaced the slice sampling codes with the lazy sampling.

mlomeli1 commented:
The paper is here https://projecteuclid.org/euclid.ss/1377696940?

mlomeli1 commented:
I think that it's best to have a look at this repo : https://github.com/BigBayes/BNPMix.jl since it is in Julia and it matches exactly the class of priors from that paper.

trappmartin commented:
@yebai I added a topic model (finite and with CRF) to the examples. Could you please take a look if what I did makes any sense. I'm not sure this is correctly implemented, particularly the LDA-CRF model. It does work but I don't quite know if the model is correct.

Thanks!
